src/Vehicle/Vehicle.cc

```diff
@@ -3286,6 +3321,10 @@ void Vehicle::_rebootCommandResultHandler(void* resultHandlerData, int /*compId*
         }
         qgcApp()->showAppMessage(tr("Vehicle reboot failed."));
     } else {
+        // here we are quite sure that device is soft rebooting
+        // closeVehicle itself is called in other scenarios, so we call persistConsumedForBatteries here
+        QMap<uint8_t, double> batteryConsumed = vehicle->collectBatteryConsumedData();
+        VehicleBatteryFactGroup::persistConsumedForBatteries(vehicle->id(), batteryConsumed);
```

Comment:
Does this mean that only soft reboot will persist the data? How about hard reboot?

Reply:
Updated so that soft reboot and connection loss persists the data. Connection loss includes hard reboot